zipkin-server/src/main/java/zipkin2/server/internal/brave/SelfTracingProperties.java

```diff
@@ -0,0 +1,62 @@
+package zipkin2.server.internal.brave;
+
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.boot.convert.DurationUnit;
+
+@ConfigurationProperties("zipkin.self-tracing")
+class SelfTracingProperties {
+
+  /** Whether self-tracing is enabled. Defaults to {@code false}. */
+  private boolean enabled = false;
+  /**
+   * The percentage of traces retained when self-tracing. If 1.0 (i.e., all traces are sampled),
+   * the value of {@link #getTracesPerSecond()} will be used for sampling traces. Defaults to
+   * {@code 1.0}, sampling all traces.
+   */
+  private float sampleRate = 1.0f;
+  /**
+   * The number of traces per second to retain. If 0, an unlimited number of traces will be
+   * retained. This value has no effect if {@link #getSampleRate()} is set to something other than
+   * {@code 1.0}. Defaults to 1 trace per second.
+   */
+  private int tracesPerSecond = 1;
```

Comment:
fyi @marcingrzejszczak @devinsba @shakuzen @llinder @zeagord @jeqo we probably want a similar thing in sleuth to default to a certain traces per second instead of 100%

Reply:
Can you create an issue for that in Sleuth?

Reply:
Sleuth defaults to 10% sampling. Do you mean changing that default sampling from 10% to 1 trace per second?